moped-etl/app/process/request.py

```diff
@@ -0,0 +1,48 @@
+#
+# Request Helper - Makes requests to a Hasura/GraphQL endpoint
+#
+
+import os
+import time
+import requests
+
+MAX_ATTEMPTS = int(os.getenv("HASURA_MAX_ATTEMPTS"))
+RETRY_WAIT_TIME = os.getenv("HASURA_RETRY_WAIT_TIME")
+HASURA_ENDPOINT = os.getenv("HASURA_ENDPOINT")
+HASURA_ADMIN_KEY = os.getenv("HASURA_ADMIN_KEY")
+
+def run_query(query):
+    """
+    Runs a GraphQL query against Hasura via an HTTP POST request.
+    :param query: string - The GraphQL query to execute (query, mutation, etc.)
+    :return: object - A Json dictionary directly from Hasura
+    """
+    # Build Header with Admin Secret
+    headers = {
+        "x-hasura-admin-secret": HASURA_ADMIN_KEY
+    }
+
+    # Try up to n times as defined by max_attempts
+    for current_attempt in range(MAX_ATTEMPTS):
+        # Try making the request via POST
+        try:
+            return requests.post(HASURA_ENDPOINT,
```

Comment:
ah, be aware that afaik `requests` will only raise an exception on connection and timeout errors. it's a good practice to use [Response.raise_for_status()](https://docs.python-requests.org/en/master/_modules/requests/models/#Response.raise_for_status) - which will throw an error based on status code ( `4xx` and `5xx`).

I'd change this line to
```python
res = requests.post(HASURA_ENDPOINT, ...)
res.raise_for_status()
return res.json()
```

Reply:
oh, right—Hasura is not going to return `4xx` errors. in any case i don't think it hurts to stick this in here, but i'm not very familiar with Hasura error behavior.